src/weighted_list.h

```diff
@@ -220,13 +220,8 @@ template <typename T> struct weighted_float_list : public weighted_list<double,
 
     protected:
 
-        size_t pick_ent( long long rand_ll ) const override {
-            double picked;
-            if( rand_ll == -1 ) {
-                picked = rng_float( nextafter( 0.0, 1.0 ), this->total_weight );
-            } else {
-                picked = ( double )rand_ll / ( double )INT_MAX * ( this->total_weight );
-            }
+        size_t pick_ent( unsigned int randi ) const override {
+            double picked = ( double )randi / ( double )RAND_MAX * ( this->total_weight );
```

Comment:
`double picked = ( double )(randi%RAND_MAX) / ( double )RAND_MAX * ( this->total_weight );`

this will make sure the picked number is <total_weight, in cases where randi>RAND_MAX because INT_MAX>RAND_MAX (such as on the windows platform where this bug was identified)